go/ct/gen/block_context_test.go

```diff
@@ -54,56 +56,476 @@ func TestBlockContextGen_Generate(t *testing.T) {
 	}
 }
 
-func TestBlockContextGen_BlockNumber(t *testing.T) {
-	istanbulBase, err := common.GetForkBlock(common.R07_Istanbul)
-	if err != nil {
-		t.Errorf("Failed to get Istanbul fork block number. %v", err)
+func TestBlockContextGen_PrintProducesConstraintFormula(t *testing.T) {
+	tests := map[string]struct {
+		setup func(*BlockContextGenerator)
+		want  string
+	}{
+		"empty": {
+			setup: func(b *BlockContextGenerator) {},
+			want:  "true",
+		},
+		"restrict-to-single-revision": {
+			setup: func(b *BlockContextGenerator) {
+				b.SetRevision(common.R07_Istanbul)
+			},
+			want: "0≤BlockNumber≤999",
+		},
+		"restrict-to-revision-range": {
+			setup: func(b *BlockContextGenerator) {
+				b.AddRevisionBounds(common.R09_Berlin, common.R11_Paris)
+			},
+			// TODO: clean up this expectation by using GetForkBlock function
+			want: "1000≤BlockNumber≤3999",
+		},
+		"restrict-to-multiple-revisions": {
+			setup: func(b *BlockContextGenerator) {
+				b.SetRevision(common.R07_Istanbul)
+				b.SetRevision(common.R09_Berlin)
+			},
+			want: "false",
+		},
+		"restrict-to-multiple-revisions-that-are-not-conflicting": {
+			setup: func(b *BlockContextGenerator) {
+				b.AddRevisionBounds(common.R07_Istanbul, common.R09_Berlin)
+				b.SetRevision(common.R07_Istanbul)
+			},
+			want: "0≤BlockNumber≤999",
+		},
+		"one-variable-in-range": {
+			setup: func(b *BlockContextGenerator) {
+				b.RestrictVariableToOneOfTheLast256Blocks("a")
+			},
+			want: "$a ∈ [BlockNumber-256..BlockNumber-1]",
+		},
+		"one-variable-out-of-range": {
+			setup: func(b *BlockContextGenerator) {
+				b.RestrictVariableToNoneOfTheLast256Blocks("a")
+			},
+			want: "$a ∉ [BlockNumber-256..BlockNumber-1]",
+		},
+		"two-variables-in-range": {
+			setup: func(b *BlockContextGenerator) {
+				b.RestrictVariableToOneOfTheLast256Blocks("a")
+				b.RestrictVariableToOneOfTheLast256Blocks("b")
+			},
+			want: "$a ∈ [BlockNumber-256..BlockNumber-1] Λ $b ∈ [BlockNumber-256..BlockNumber-1]",
+		},
+		"one-variable-with-fixed-value": {
```

Comment:
nit: (also for other places)
```suggestion
		"one-variable-with-fixed-offset": {
```